Youtube Scraper Tests Fail

- Issue type: Bug report

### Short description

Due to recent modifications in Youtube UI, the tests are failing.

```
> Task :test

org.loklak.TestRunner > org.loklak.harvester.YoutubeScraperTest.parseFromBufferedReaderTest FAILED
    java.lang.AssertionError at YoutubeScraperTest.java:68

org.loklak.harvester.YoutubeScraperTest > parseFromBufferedReaderTest FAILED
    java.lang.AssertionError at YoutubeScraperTest.java:68

31 tests completed, 2 failed


FAILURE: Build failed with an exception.
```

--- a/src/org/loklak/harvester/YoutubeScraper.java
+++ b/src/org/loklak/harvester/YoutubeScraper.java
@@ -1,23 +1,25 @@
 /**
- *  YoutubeScraper
- *  Copyright 22.03.2016 by Michael Peter Christen, @0rb1t3r
+ * YoutubeScraper
+ * Copyright 22.03.2016 by Michael Peter Christen, @0rb1t3r
  *
- *  This library is free software; you can redistribute it and/or
- *  modify it under the terms of the GNU Lesser General Public
- *  License as published by the Free Software Foundation; either
- *  version 2.1 of the License, or (at your option) any later version.
- *  
- *  This library is distributed in the hope that it will be useful,
- *  but WITHOUT ANY WARRANTY; without even the implied warranty of
- *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
- *  Lesser General Public License for more details.
- *  
- *  You should have received a copy of the GNU Lesser General Public License
- *  along with this program in the file lgpl21.txt
- *  If not, see <http://www.gnu.org/licenses/>.
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program in the file lgpl21.txt
+ * If not, see <http://www.gnu.org/licenses/>.
  */
 
 package org.loklak.harvester;
+
+import com.google.common.base.CharMatcher;
 
 import java.io.BufferedReader;
 import java.io.File;
@@ -31,19 +33,18 @@ import java.nio.charset.StandardCharsets;
 import java.util.Map;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
-import com.google.common.base.CharMatcher;
+
 import org.apache.http.client.utils.URIBuilder;
 import org.json.JSONArray;
 import org.json.JSONException;
 import org.json.JSONObject;
-import org.loklak.tools.CharacterCoding;
 import org.loklak.data.DAO;
-import org.loklak.harvester.BaseScraper;
-import org.loklak.harvester.Post;
 import org.loklak.objects.Timeline2;
 import org.loklak.server.BaseUserRole;
+import org.loklak.tools.CharacterCoding;
 
-public class YoutubeScraper extends BaseScraper{
+
+public class YoutubeScraper extends BaseScraper {
 
     //TODO: Not used right now, commenting temporarily, to work on this
     //public final static ExecutorService executor = Executors.newFixedThreadPool(40);
@@ -89,7 +90,7 @@ public class YoutubeScraper extends BaseScraper{
         URIBuilder url = null;
         String midUrl = "search/";
         try {
-            switch(type) {
+            switch (type) {
                 case "search":
                     midUrl = "search/";
                     url = new URIBuilder(this.baseUrl + midUrl);
@@ -116,7 +117,7 @@ public class YoutubeScraper extends BaseScraper{
     }
 
     @Override
-     public Post getDataFromConnection() throws IOException {
+    public Post getDataFromConnection() throws IOException {
         String url = this.prepareSearchUrl(this.type);
         return getDataFromConnection(url, this.type);
     }
@@ -126,18 +127,18 @@ public class YoutubeScraper extends BaseScraper{
         this.query = this.getExtraValue("query");
         //TODO: Add scraper for search and user
         type = "url";
-        if("".equals(type)) {
+        if ("".equals(type)) {
             type = "search";
-        } else if("url".equals(type)) {
-            if(this.query.contains("youtube.com/results")) {
+        } else if ("url".equals(type)) {
+            if (this.query.contains("youtube.com/results")) {
                 this.type = "search";
-            } else if(this.query.contains("youtube.com/watch")) {
+            } else if (this.query.contains("youtube.com/watch")) {
                 this.type = "video";
                 this.query = this.query.substring(this.query.length() - 11);
-            } else if(this.query.contains("youtube.com/channel")) {
+            } else if (this.query.contains("youtube.com/channel")) {
                 this.type = "user";
                 this.query = this.query.substring(this.query.length() - 24);
-            } else if(this.query.contains("youtu.be")) {
+            } else if (this.query.contains("youtu.be")) {
                 this.query = this.query.substring(this.query.length() - 11);
             }
         }
@@ -148,7 +149,7 @@ public class YoutubeScraper extends BaseScraper{
     protected Post scrape(BufferedReader br, String type, String url) {
         Post out = new Post(true);
         Timeline2 postList = new Timeline2(this.order);
-        switch(type) {
+        switch (type) {
             case "user":
                 //TODO: Add scraper
                 break;
@@ -172,166 +173,171 @@ public class YoutubeScraper extends BaseScraper{
             FileInputStream fis = new FileInputStream(file);
             json = this.parseVideo(fis, type, url);
             fis.close();
-        } catch(FileNotFoundException e) {
+        } catch (FileNotFoundException e) {
             DAO.severe(e);
-        } catch(IOException e) { 
+        } catch (IOException e) {
             DAO.severe("Connection error for query = " + this.query);
         }
         return json;
     }
-    
+
     public Post parseVideo(InputStream is, String type, String url) {
         Post json = new Post();
         try {
             BufferedReader reader = new BufferedReader(new InputStreamReader(is, StandardCharsets.UTF_8));
             json = this.parseVideo(reader, type, url);
             reader.close();
-        } catch(IOException e) { 
+        } catch (IOException e) {
             DAO.severe("Connection error for query = " + this.query);
         }
 
         return json;
     }
 
-    public Post parseVideo(final BufferedReader br, String type, String url){
+    public Post parseVideo(final BufferedReader br, String type, String url) {
         String input = "";
         Post json = new Post(true);
         boolean parse_span = false, parse_license = false;
         // values for span
-        String itemprop= "";
-        String itemtype = "";
+        String itemProp = "";
+        String itemType = "";
         try {
-        while ((input = br.readLine()) != null) {
-            input = input.trim();
-            int p;
+            while ((input = br.readLine()) != null) {
+                input = input.trim();
+                int p;
 
-            if (parse_license) {
-                if ((p = input.indexOf("<li")) >= 0) {
-                    String tag = parseTag(input, p);
-                    if (tag == null) continue;
-                    if (tag.startsWith("<a ")) {
-                        tag = parseTag(tag, 0);
-                        addRDF(new String[]{"youtube", "category", tag}, json);
-                    } else {
-                        addRDF(new String[]{"youtube", "license", tag}, json);
+                if (parse_license) {
+                    if ((p = input.indexOf("<li")) >= 0) {
+                        String tag = parseTag(input, p);
+                        if (tag == null) continue;
+                        if (tag.startsWith("<a ")) {
+                            tag = parseTag(tag, 0);
+                            addRDF(new String[]{"youtube", "category", tag}, json);
+                        } else {
+                            addRDF(new String[]{"youtube", "license", tag}, json);
+                        }
+                        parse_license = false;
                     }
-                    parse_license = false;
-                    continue;
-                }
-            } else if (parse_span) {
-                if ((p = input.indexOf("itemprop=\"")) >= 0) {
-                    String[] token = parseItemprop(input, p, new String[]{"href", "content"}, "");
-                    if (token == null) continue;
-                    int q = itemtype.indexOf("//"); if (q < 0) continue;
-                    String subject = itemtype.substring(q + 2).replace('.', '_').replace('/', '_');
-                    String predicate = itemprop + "_" + token[1];
-                    String object = token[2];
-                    addRDF(new String[]{subject, predicate, object}, json);
-                    continue;
-                }
-                if (input.indexOf("</span>") >= 0) {
-                    parse_span = false;
-                    continue;
-                }
-            } else {
-                tags: for (String tag: html_tags) {
-                    if ((p = input.indexOf("<" + tag)) >= 0) {
-                        addRDF(new String[]{"html", tag, parseTag(input, p)}, json);
-                        continue tags;
+                } else if (parse_span) {
+                    if ((p = input.indexOf("itemprop=\"")) >= 0) {
+                        String[] token = parseItemprop(input, p, new String[]{"href", "content"}, "");
+                        if (token == null) continue;
+                        int q = itemType.indexOf("//");
+                        if (q < 0) continue;
+                        String subject = itemType.substring(q + 2).replace('.', '_').replace('/', '_');
+                        String predicate = itemProp + "_" + token[1];
+                        String object = token[2];
+                        addRDF(new String[]{subject, predicate, object}, json);
+                        continue;
                     }
-                }
-                vocs: for (String subject: microformat_vocabularies) {
-                    if ((p = input.indexOf("property=\"" + subject + ":")) >= 0) {
-                        addRDF(parseMicroformat(input, "property", p), json);
-                        continue vocs;
+                    if (input.contains("</span>")) {
+                        parse_span = false;
                     }
-                    if ((p = input.indexOf("name=\"" + subject + ":")) >= 0) {
-                        addRDF(parseMicroformat(input, "name", p), json);
-                        continue vocs;
+                } else {
+                    tags:
+                    for (String tag : html_tags) {
+                        if ((p = input.indexOf("<" + tag)) >= 0) {
+                            addRDF(new String[]{"html", tag, parseTag(input, p)}, json);
+                            continue tags;
+                        }
                     }
-                }
-                if ((p = input.indexOf("span itemprop=\"")) >= 0) {
-                    String[] token = parseItemprop(input, p, new String[]{"itemtype"}, "");
-                    if (token == null) continue;
-                    itemprop = token[1];
-                    itemtype = token[2];
-                    parse_span = true;
-                    continue;
-                }
-                if ((p = input.indexOf("itemprop=\"")) >= 0) {
-                    String[] token = parseItemprop(input, p, new String[]{"content"}, "youtube");
-                    if (token == null) continue;
-                    addRDF(token, json);
-                    continue;
-                }
-                if ((p = input.indexOf("class=\"content watch-info-tag-list")) >= 0) {
-                    parse_license = true;
-                    continue;
-                }
-                if ((p = input.indexOf("yt-subscriber-count")) >= 0) {
-                    String subscriber_string = parseProp(input, p, "title");
-                    if (subscriber_string == null) continue;
-                    json.put("youtube_subscriber", parseValue(subscriber_string));
-                    continue;
-                }
-                if (input.indexOf("\"like this") > 0 && (p = input.indexOf("yt-uix-button-content")) >= 0) {
-                    String likes_string = parseTag(input, p);
-                    json.put("youtube_likes", parseValue(likes_string));
-                    continue;
-                }
-                if (input.indexOf("\"dislike this") > 0 && (p = input.indexOf("yt-uix-button-content")) >= 0) {
-                    String dislikes_string = parseTag(input, p);
-                    json.put("youtube_dislikes", parseValue(dislikes_string));
-                    continue;
-                }
-                if ((p = input.indexOf("watch-view-count")) >= 0) {
-                    String viewcount_string = parseTag(input, p);
-                    if (viewcount_string == null) continue;
-                    viewcount_string = viewcount_string.replace(" views", "");
-                    if (viewcount_string.length() == 0) continue;
-                    json.put("youtube_viewcount", viewcount_string);
-                    continue;
-                }
-                if ((p = input.indexOf("watch?v=")) >= 0) {
-                    p += 8;
-                    int q = input.indexOf("\"", p);
-                    if (q > 0) {
-                        String videoid = input.substring(p, q);
-                        int r = videoid.indexOf('&');
-                        if (r > 0) videoid = videoid.substring(0, r);
-                        addRDF(new String[]{"youtube", "next", videoid}, json);
+                    vocs:
+                    for (String subject : microformat_vocabularies) {
+                        if ((p = input.indexOf("property=\"" + subject + ":")) >= 0) {
+                            addRDF(parseMicroformat(input, "property", p), json);
+                            continue vocs;
+                        }
+                        if ((p = input.indexOf("name=\"" + subject + ":")) >= 0) {
+                            addRDF(parseMicroformat(input, "name", p), json);
+                            continue vocs;
+                        }
+                    }
+                    if ((p = input.indexOf("span itemprop=\"")) >= 0) {
+                        String[] token = parseItemprop(input, p, new String[]{"itemtype"}, "");
+                        if (token == null) continue;
+                        itemProp = token[1];
+                        itemType = token[2];
+                        parse_span = true;
+                        continue;
+                    }
+                    if ((p = input.indexOf("itemprop=\"")) >= 0) {
+                        String[] token = parseItemprop(input, p, new String[]{"content"}, "youtube");
+                        if (token == null) continue;
+                        addRDF(token, json);
+                        continue;
+                    }
+                    if ((p = input.indexOf("class=\"content watch-info-tag-list")) >= 0) {
+                        parse_license = true;
+                        continue;
+                    }
+                    if ((p = input.indexOf("yt-subscriber-count")) >= 0) {
+                        String subscriber_string = parseProp(input, p, "title");
+                        if (subscriber_string == null) continue;
+                        json.put("youtube_subscriber", parseValue(subscriber_string));
+                        continue;
+                    }
+                    if (input.indexOf("\"like this") > 0 && (p = input.indexOf("yt-uix-button-content")) >= 0) {
+                        String likes_string = parseTag(input, p);
+                        json.put("youtube_likes", parseValue(likes_string));
+                        continue;
+                    }
+                    if (input.indexOf("\"dislike this") > 0 && (p = input.indexOf("yt-uix-button-content")) >= 0) {
+                        String dislikes_string = parseTag(input, p);
+                        json.put("youtube_dislikes", parseValue(dislikes_string));
+                        continue;
+                    }
+                    if ((p = input.indexOf("watch-view-count")) >= 0) {
+                        String viewcount_string = parseTag(input, p);
+                        if (viewcount_string == null) continue;
+                        viewcount_string = viewcount_string.replace(" views", "");
+                        if (viewcount_string.length() == 0) continue;
+                        json.put("youtube_viewcount", viewcount_string);
+                        continue;
+                    }
+                    if ((p = input.indexOf("watch?v=")) >= 0) {
+                        p += 8;
+                        int q = input.indexOf("\"", p);
+                        if (q > 0) {
+                            String videoid = input.substring(p, q);
+                            int r = videoid.indexOf('&');
+                            if (r > 0) videoid = videoid.substring(0, r);
+                            addRDF(new String[]{"youtube", "next", videoid}, json);
+                            continue;
+                        }
+                    }
+                    if ((p = input.indexOf("playlist-header-content")) >= 0) {
+                        String playlist_title = parseProp(input, p, "data-list-title");
+                        if (playlist_title == null) continue;
+                        addRDF(new String[]{"youtube", "playlist_title", playlist_title}, json);
+                        continue;
+                    }
+                    if ((p = input.indexOf("yt-uix-scroller-scroll-unit")) >= 0) {
+                        String playlist_videoid = parseProp(input, p, "data-video-id");
+                        if (playlist_videoid == null) continue;
+                        addRDF(new String[]{"youtube", "playlist_videoid", playlist_videoid}, json);
+                        continue;
+                    }
+                    if ((p = input.indexOf("watch-description-text")) >= 0) {
+                        p = input.indexOf('>', p);
+                        int q = input.indexOf("</div", p);
+                        String text = input.substring(p + 1, q < 0 ? input.length() : q);
+                        text = paragraph.matcher(brend.matcher(text).replaceAll("\n")).replaceAll("").trim();
+                        Matcher m;
+                        anchor_loop:
+                        while ((m = anchor_pattern.matcher(text)).find()) try {
+                            text = m.replaceFirst(m.group(1) + " ");
+                        } catch (IllegalArgumentException e) {
+                            text = "";
+                            break anchor_loop;
+                        }
+                        text = CharacterCoding.html2unicode(text);
+                        json.put("youtube_description", text);
                         continue;
                     }
                 }
-                if ((p = input.indexOf("playlist-header-content")) >= 0) {
-                    String playlist_title = parseProp(input, p, "data-list-title");
-                    if (playlist_title == null) continue;
-                    addRDF(new String[]{"youtube", "playlist_title", playlist_title}, json);
-                    continue;
-                }
-                if ((p = input.indexOf("yt-uix-scroller-scroll-unit")) >= 0) {
-                    String playlist_videoid = parseProp(input, p, "data-video-id");
-                    if (playlist_videoid == null) continue;
-                    addRDF(new String[]{"youtube", "playlist_videoid", playlist_videoid}, json);
-                    continue;
-                }
-                if ((p = input.indexOf("watch-description-text")) >= 0) {
-                    p = input.indexOf('>', p);
-                    int q = input.indexOf("</div", p);
-                    String text = input.substring(p + 1, q < 0 ? input.length() : q);
-                    text = paragraph.matcher(brend.matcher(text).replaceAll("\n")).replaceAll("").trim();
-                    Matcher m;
-                    anchor_loop: while ((m = anchor_pattern.matcher(text)).find()) try {
-                        text = m.replaceFirst(m.group(1) + " ");
-                    } catch (IllegalArgumentException  e) {text = ""; break anchor_loop;}
-                    text = CharacterCoding.html2unicode(text);
-                    json.put("youtube_description", text);
-                    continue;
-                }
             }
-        }
             br.close();
-        } catch(IOException e) {
+        } catch (IOException e) {
             json.put("error", "Connection error while fetching");
             json.put("search_url", url);
         } catch (Throwable e) {
@@ -342,7 +348,7 @@ public class YoutubeScraper extends BaseScraper{
 
         return json;
     }
-    
+
     private String parseValue(String n) {
         String number_value = CharMatcher.WHITESPACE.removeFrom(n);
         Pattern number_format = Pattern.compile("^(\\d{1,3})(((,\\d{3})*)|((\\.\\d+)?[MBK]))?$");
@@ -352,18 +358,25 @@ public class YoutubeScraper extends BaseScraper{
         }
         return number_value;
     }
-    
+
     private final static Pattern numberfix = Pattern.compile(",|\\.");
     private final static Pattern paragraph = Pattern.compile("<p.*>|</p.*>");
     private final static Pattern brend = Pattern.compile("<br />");
     private final static Pattern anchor_pattern = Pattern.compile("<a .*?>(.*?)</a>");
-    
+
     private String[] parseMicroformat(String line, String key, int start) {
-        int p  = line.indexOf(key + "=\"", start); if (p < 0) return null; p += key.length() + 2;
-        int c  = line.indexOf(":", p); if (c < 0) return null;
-        int q  = line.indexOf("\"", c); if (q < 0) return null;
-        int r  = line.indexOf("content=\"", q); if (r < 0) return null; r += 9;
-        int s  = line.indexOf("\"", r); if (s < 0) return null;
+        int p = line.indexOf(key + "=\"", start);
+        if (p < 0) return null;
+        p += key.length() + 2;
+        int c = line.indexOf(":", p);
+        if (c < 0) return null;
+        int q = line.indexOf("\"", c);
+        if (q < 0) return null;
+        int r = line.indexOf("content=\"", q);
+        if (r < 0) return null;
+        r += 9;
+        int s = line.indexOf("\"", r);
+        if (s < 0) return null;
         // this is a rdf statement
         String subject = line.substring(p, c).replace(':', '_');
         String predicate = line.substring(c + 1, q).replace(':', '_');
@@ -372,23 +385,28 @@ public class YoutubeScraper extends BaseScraper{
     }
 
     private String[] parseItemprop(String line, int start, String[] objectnames, String subject) {
-        int p  = line.indexOf("itemprop=\"", start); if (p < 0) return null; p += 10;
-        int q  = line.indexOf("\"", p); if (q < 0) return null;
+        int p = line.indexOf("itemprop=\"", start);
+        if (p < 0) return null;
+        p += 10;
+        int q = line.indexOf("\"", p);
+        if (q < 0) return null;
         int r = -1;
-        objectscan: for (String objectname: objectnames) {
-            r  = line.indexOf(objectname + "=\"", q);
+        objectscan:
+        for (String objectname : objectnames) {
+            r = line.indexOf(objectname + "=\"", q);
             if (r < 0) continue objectscan;
             r += objectname.length() + 2;
             break;
         }
         if (r < 0) return null;
-        int s  = line.indexOf("\"", r); if (s < 0) return null;
+        int s = line.indexOf("\"", r);
+        if (s < 0) return null;
         // this becomes a rdf statement
         String predicate = line.substring(p, q).replace(':', '_');
         String object = line.substring(r, s);
         return new String[]{subject, predicate, object};
     }
-    
+
     private void addRDF(String[] spo, Post post) {
         if (spo == null) return;
         String subject = spo[0];
@@ -404,15 +422,15 @@ public class YoutubeScraper extends BaseScraper{
             post.put(key, objects);
         }
         // double-check (wtf why is ths that complex?)
-        for (Object o: objects) {
+        for (Object o : objects) {
             if (o instanceof String && ((String) o).equals(object)) return;
         }
         // add the object to the objects
         objects.put(object);
     }
-    
+
     private String parseProp(String line, int start, String key) {
-        int p  = line.indexOf(key + "=\"", start);
+        int p = line.indexOf(key + "=\"", start);
         if (p > 0) {
             int q = line.indexOf('"', p + key.length() + 2);
             if (q > 0) {
@@ -421,7 +439,7 @@ public class YoutubeScraper extends BaseScraper{
         }
         return null;
     }
-    
+
     private String parseTag(String line, int start) {
         int p = line.indexOf('>', start);
         if (p < 0) return null;
@@ -431,7 +449,8 @@ public class YoutubeScraper extends BaseScraper{
             char a = line.charAt(q);
             if (a == '<') {
                 if (line.charAt(q + 1) != 'i') {
-                    if (line.charAt(q + 1) == '/') c--; else c++;
+                    if (line.charAt(q + 1) == '/') c--;
+                    else c++;
                 }
             }
             q++;
@@ -439,5 +458,5 @@ public class YoutubeScraper extends BaseScraper{
         if (c != 0) return "";
         return line.substring(p + 1, q - 1).trim();
     }
-    
+
 }

--- a/test/org/loklak/harvester/YoutubeScraperTest.java
+++ b/test/org/loklak/harvester/YoutubeScraperTest.java
@@ -1,16 +1,14 @@
 package org.loklak.harvester;
 
-
-import org.json.JSONObject;
-import org.junit.Test;
-
 import java.io.BufferedReader;
 import java.io.IOException;
 import java.io.InputStream;
+import java.io.InputStreamReader;
 import java.net.URL;
-import org.loklak.data.DAO;
-import org.loklak.http.ClientConnection;
+import java.nio.charset.StandardCharsets;
 
+import org.junit.Test;
+import org.loklak.data.DAO;
 import static org.hamcrest.core.Is.is;
 import static org.junit.Assert.assertThat;
 
@@ -18,13 +16,14 @@ import static org.junit.Assert.assertThat;
  * @author evrnsky
  * @version 0.1
  * @since 17.04.2017
- *
+ * <p>
  * This is unit test for YoutubeScrapper.
  */
 public class YoutubeScraperTest {
 
     /**
      * When try parse video from input stream should check that video parsed.
+     *
      * @throws IOException if some problem with open stream for reading data.
      */
     @Test
@@ -35,37 +34,35 @@ public class YoutubeScraperTest {
 
         try {
             fis = new URL(url).openStream();
-
+            Post video = ytubeScrape.parseVideo(fis, "url", url);
+            DAO.log(video.toString());
+            assertThat(video.get("html_title").toString(), is("[\"Iggy Azalea - Team (Explicit) - YouTube\"]"));
         } catch (IOException e) {
             DAO.log("YoutubeScraperTest.parseFromInputStreamTest() failed to connect to network. url:" + url);
         }
 
-        JSONObject video = ytubeScrape.parseVideo(fis, "url", url);
-        assertThat(video.get("html_title").toString(), is("[\"Iggy Azalea - Team (Explicit) - YouTube\"]"));
     }
 
     /**
      * When try parse video from buffered reader should check that method return valid json.
+     *
      * @throws IOException if some error happened with open stream for reading data.
      */
     @Test
     public void parseFromBufferedReaderTest() throws IOException {
         YoutubeScraper ytubeScrape = new YoutubeScraper();
         String url = "https://www.youtube.com/watch?v=KVGRN7Z7T1A";
-        BufferedReader br = null;
 
         try {
-            ClientConnection connection = new ClientConnection(url);
             //Check Network issue
-            assertThat(connection.getStatusCode(), is(200));
-
-            br = ytubeScrape.getHtml(connection);
+            BufferedReader br = new BufferedReader(new InputStreamReader((new URL(url)).openStream(), StandardCharsets.UTF_8));
+            Post video = ytubeScrape.parseVideo(br, "url", "https://www.youtube.com/watch?v=KVGRN7Z7T1A");
+            DAO.log(video.toString());
+            assertThat(video.get("html_title").toString(), is("[\"Iggy Azalea - Team (Explicit) - YouTube\"]"));
         } catch (IOException e) {
             DAO.log("YoutubeScraperTest.parseFromBufferedReaderTest()() failed to connect to network. url:" + url);
         }
 
-        JSONObject video = ytubeScrape.parseVideo(br, "url", "https://www.youtube.com/watch?v=KVGRN7Z7T1A");
-        assertThat(video.get("html_title").toString(), is("[\"Iggy Azalea - Team (Explicit) - YouTube\"]"));
     }
 
 }

